packages/samples/headless-react/src/components/triggers/notify-trigger.fn.tsx

```diff
@@ -0,0 +1,31 @@
+import {useEffect, useState, FunctionComponent} from 'react';
+import {NotifyTrigger as HeadlessNotifyTrigger} from '@coveo/headless';
+
+interface HeadlessNotifyTriggerProps {
+  controller: HeadlessNotifyTrigger;
+}
+
+export const NotifyTrigger: FunctionComponent<HeadlessNotifyTriggerProps> = (
+  props
+) => {
+  const {controller} = props;
+  const [state, setState] = useState(controller.state);
+
+  useEffect(() => controller.subscribe(() => updateState()), []);
+
+  const updateState = () => {
+    setState(props.controller.state);
+  };
+
+  return <div>The notification is: {state.notify}</div>;
```

Comment:
Instead of rendering something, perhaps the notify trigger samples could `alert()`. I believe this would demonstrate well that the subscriber is only called when there's a notification and doesn't get redundantly called.